### DISCUSSION TITLE
What is the target audience?

### DISCUSSION THREAD

**Discussion by xipengxiao** [becarpenter/book6#4] [General]
•	What is the target audience?  It looks like for graduate students/end users.  Is that right?  I think we need a book for graduate students/end users, but we also need another book for network administrators.  For the former, the book can be 200-300 pages like a textbook. For the latter, we need to limit it within 100 pages.  The focus of the books will be different too.

- If you are targeting the graduate students/end users, I may move to lead the book for network administrators.  But I can still contribute to your book, for example the "deployment" chapter or some sections in it.  My github ID is: xipengxiao.  
- In case I am to lead a separate book for network admins, is it OK to use some content in your book, e.g. the “IPv6 basic technology”?  Now my thinking is, let everybody who has a book idea propose their TOC on github, and let v6ops people choose which book(s) they want to contribute.  This “voting” process will enable us to select 1-2 books to work on, and I hope that the selected book owners can share their content.  Do you think this makes sense?  

•	If the target audience is network admins, a "Recent Progress in IPv6" section/chapter at the beginning of the book is likely needed.  

- The reason is, many people consider IPv6 an old and boring topic.  For example, even many of my Huawei colleagues hold this view.  They would say, we talked about IPv6 10 years ago, and it didn't happen, why are you still talking about it?  My experience is, when speaking to an audience in the networking industry about IPv6, if I don't immediately tell them that IPv6 has made huge progress in the last 5 years and is not that "old IPv6" in their mind, they will think that I am selling them "cold turkey" and their mind will wander away.  
- But with the current "Introduction" chapter, I don't quite know how to fit this "Recent Progress in IPv6" section/chapter in.  One possibility is, we fundamentally change the "introduction" chapter, deleting the 3 "view from the top" section, and put in the "Recent Progress in IPv6".  The rationale is, although I think it a good idea to talk about IPv6 from the end users/programmers' perspective, I suspect that we have nothing positive to say - other than more works/troubles, what will end users/programmers get out of IPv6 whose primary benefit is more address? 
- If the target audience is graduate students/end users, you may not need a "Recent Progress in IPv6" section/chapter

**Comment by becarpenter** [comment]
I think that if we are clever, we can do both in one book. That is why I included "How to use this book (alternative learning paths)" in the draft list of contents. We could provide two (or more) paths through the book by using  links...
Certainly we want to convey that IPv6 in 2022 is a finished product - I think the message is even stronger than "Recent Progress", it's "IPv6 is a mature protocol used for more than 30% of the Internet traffic".
About the user view, indeed there is not much to say: it's just the Internet! There is something to say for the programmer, however. Unfortunately it is impossible to write a robust application without correct handling of dual stack issues.

**Reply by joaolucasmacedo** [reply]
> I think that if we are clever, we can do both in one book. That is why I included "How to use this book (alternative learning paths)" in the draft list of contents. We could provide two (or more) paths through the book by using links... Certainly we want to convey that IPv6 in 2022 is a finished product - I think the message is even stronger than "Recent Progress", it's "IPv6 is a mature protocol used for more than 30% of the Internet traffic". About the user view, indeed there is not much to say: it's just the Internet! There is something to say for the programmer, however. Unfortunately it is impossible to write a robust application without correct handling of dual stack issues.

I totally agree with you Brian. I think we can do both approaches. 
I suggest that a practical approach is key. I've see many IPv6 books and some topics on world relevant books, like Tanenbaum, Kurose, Stallings, and they give just a very brief explanation on that topic. Also, other books smaller books have a good text on IPv6 but they present the topic like a "wikipedia article", as if IPv6 is disconnected from the actual/current network operation environment. There are other nice books, like Hughes, but they are kind outdated.

**Reply by becarpenter** [reply]
I also thought yesterday that we might add a chapter on "Legacy solutions" where we would list (without detail) abandoned or failed mechanisms (site local, 6to4, Teredo, modified EUI64, etc.). The reason is to prevent people wasting time or being misled by older RFCs or books.

**Comment by becarpenter** [comment]
@xipengxiao, please feel free to suggest new sections in the Contents  and changes to the text in the [Introduction](https://github.com/becarpenter/book6/blob/main/1.%20Introduction%20and%20Foreword/1.%20Introduction.md) to ensure that we can meet your goals.

**Comment by timchown** [comment]
> On 5 Sep 2022, at 21:42, Brian E Carpenter ***@***.***> wrote:
> 
> I also thought yesterday that we might add a chapter on "Legacy solutions" where we would list (without detail) abandoned or failed mechanisms (site local, 6to4, Teredo, modified EUI64, etc.). The reason is to prevent people wasting time or being misled by older RFCs or books.

Can we put SRv6 in there yet? ;)

But a nice idea. 

Tim

**Comment by Paolo-Volpato** [comment]
Hi All,
I am using this thread since it's already available.
My colleague Giuseppe and I are writing a contribution for the "Deployment" section. 
The contribution will be based on the work we did with the "IPv6 Deployment Status" draft in v6ops.

In that draft we mentioned the transition to "IPv6-only" networks and services.
We wonder if the section on "Coexistence with legacy IPv4" is the right place or is it better to add a new section?

Giuseppe and Paolo

**Comment by becarpenter** [comment]
I think there will be cross-references between those topics in any structure. I recommend writing fairly short sections (headed with ## Topic in a separate markdown file) and then we can re-combine them later in the best way. In general I think short sections in individual files is best for us.
Personally I think that the various meanings of "IPv6 only" probably fit naturally into the coexistence chapter.  I do like the terminology explanations in draft-ietf-v6ops-ipv6-deployment.

**Comment by timchown** [comment]
> On 13 Sep 2022, at 06:52, Brian E Carpenter ***@***.***> wrote:
> 
> I think there will be cross-references between those topics in any structure. I recommend writing fairly short sections (headed with ## Topic in a separate markdown file) and then we can re-combine them later in the best way. In general I think short sections in individual files is best for us.
> Personally I think that the various meanings of "IPv6 only" probably fit naturally into the coexistence chapter. I do like the terminology explanations in draft-ietf-v6ops-ipv6-deployment.

Given this is a “living book”, maybe it’s a place where the common issues of running IPv6-only, e.g. for users o a campus WiFi, could be maintained.  I think there are some places that track this, not sure how up to date, but previously there was no option to capture these alongside the theory.  But that could be done here.

Tim

**Comment by buraglio** [comment]
Along those lines I am planning to add the lessons learned from our multi-year transition into an IPv6-only management network for our international backbone. It touches almost all of the edge cases and, I think, is fairly unique compared to a lot of the info out there that focuses on end stations.

**Comment by becarpenter** [comment]
We should get someone in the LHC networking world to contribute too.

**Comment by timchown** [comment]
> On 13 Sep 2022, at 21:11, Brian E Carpenter ***@***.***> wrote:
> 
> We should get someone in the LHC networking world to contribute too.

The most natural person for that would be Dave Kelsey, who chairs the WLCG HEPiX IPv6 WG.

Are you considering some “Deployment in practice” sections/chapters?

Tim

**Comment by becarpenter** [comment]
Dave, of course. I'll write to him in a few days. I'm just focussed on https://github.com/becarpenter/book6/discussions/12#discussioncomment-3602052 first.

We have both a Deployment chapter and a Case Studies chapter planned. It could go in either.
